20231804T160439/index.md

```diff
@@ -0,0 +1,394 @@
+---
+title: 'Relacionamento das bases no relatório operacional: método Concatenate x Linktable'
+---
+
+# Introdução 
+
+As bases  que compõem o relatório operacional, painel de dados de uso na SPLOR que utiliza a ferramenta de BI qlikview, são provenientes do [armazém B.O](http://www.armazem.mg.gov.br/), universos SIAFI e SIAD, e também das bases de Restimativa de Despesa e Receita, que atualmente se encontram compartilhadas no OneDrive e são periodicamente atualizadas pelas equipes da DCMEFO e DCAF. 
+
+Essas bases têm como característica apresentar diferentes granularidades (dimensões) nos dados entre elas, em razão das especificidades inerentes a natureza dos fluxos a que pertencem. E como será mostrado ao longo desta nota, isso acarreta em um desafio adicional ao usuário desenvolvedor do qlikview na tarefa de relacioná-las e gerar visualizações.
+
+O objetivo desta nota é apresentar as diferentens granularidades de cada uma das bases, bem como comparar o método de empilhamento das bases (mais intuitivo a usuários iniciantes), com o  método [linktable](QlikView_Technical_Brief-Concatenate-and-Link-tables.pdf), solução encontrada para relacionar as bases do retório operacional.
+
+# As diferentes granularidades das bases que alimentam o relatório operacional qlikview:
+
+O qlikview é uma ferramenta de selfservice BI, que permite que usuários com conhecimento de comandos básicos da ferramenta sejam capazes de criar suas próprias visualizações. A forma mais intuitiva, e geralmente mais a utilizada por esses usuários, é empilhar todas as bases que se deseja utilizar, por meio do comando `Concatenate` no `Script` do aplicativo. Este método é satisfatório quando inexiste diferença de granularidade nas bases. Entretanto, no caso das bases do relatório operacional, que apresentam diferentes granularidades entre suas bases, o empilhamento em uma única base traz problemas ao usuário da ferramenta, ilustrados no tópico a seguir.
+
+As diferentes granularidades (dimensões) dos dados pode ser verificada por meio do ementário completo das bases do projeto disponível em [data package relatorio operacional](https://gist.github.com/hslinhares/68a3d06eae13b8facb1df42e1095c49e).
+
+Para exemplificar os métodos de empilhamento (concatenate) e linktable nesta nota, são utilizados como amostra, as bases de Execução da Despesa, Crédito Inicial e Autorizado, e Aprovação de Cota Orçamentária, todas do exercício 2022 e provenientes do armazém B.O. 
+
+# Empilhamento das bases (método concatenate) no qlikview:
+
+Acessando o `Script` do qlikview, por meio do `ctrl E`, os seguintes passos devem ser seguidos:
+
+## Passo 1: Empilhar as bases mensais da Execução da Despesa 2022:
+
+```default
+//cria uma tabela temporária com a execução de todos os meses de 2022.
+
+execucao_temp:
+LOAD ANO, 
```

Comment:
Acho que vale a pena reduzir o número de colunas inseridas pra facilitar a visualização tanto dos scripts quanto das imagens.

Isso inclusive facilita se valer a pena criar alguns exemplos ilustrativos dos dados em si, tipo esse do livro [Star Schema The Complete Reference](https://www.amazon.com/Schema-Complete-Reference-Christopher-Adamson/dp/0071744320):

![image](https://github.com/splor-mg/notas/assets/4969445/e5b86ef3-d463-480e-8de2-adbb24e3cc1e)